awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -0,0 +1,372 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import env_fallback
+from ansible.module_utils.urls import Request, SSLValidationError, ConnectionError
+from ansible.module_utils.six.moves.urllib.parse import urlparse, urlencode
+
+from socket import gethostbyname
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+from ansible.module_utils.six.moves.http_cookiejar import CookieJar
+from ansible.module_utils.six.moves.configparser import ConfigParser, NoOptionError, NoSectionError
+import re
+from json import loads, dumps
+from os.path import isfile
+from os import access, R_OK
+
+
+class TowerModule(AnsibleModule):
+    url = None
+    honorred_settings = ['host', 'username', 'password', 'verify_ssl', 'oauth_token']
+    host = '127.0.0.1'
+    username = None
+    password = None
+    verify_ssl = True
+    oauth_token = None
+    oauth_token_id = None
+    session = None
+    cookie_jar = CookieJar()
+    authenticated = False
+    json_output = {'changed': False}
+
+    def __init__(self, argument_spec, **kwargs):
+        args = dict(
+            tower_host=dict(required=False, fallback=(env_fallback, ['TOWER_HOST'])),
+            tower_username=dict(required=False, fallback=(env_fallback, ['TOWER_USERNAME'])),
+            tower_password=dict(no_log=True, required=False, fallback=(env_fallback, ['TOWER_PASSWORD'])),
+            validate_certs=dict(type='bool', aliases=['tower_verify_ssl'], required=False, fallback=(env_fallback, ['TOWER_VERIFY_SSL'])),
+            tower_oauthtoken=dict(type='str', no_log=True, required=False, fallback=(env_fallback, ['TOWER_OAUTH_TOKEN'])),
+            tower_config_file=dict(type='path', required=False, default=None),
+        )
+        args.update(argument_spec)
+        kwargs['supports_check_mode'] = True
+
+        super(TowerModule, self).__init__(argument_spec=args, **kwargs)
+
+        # If we have a tower config, load it
+        if self.params.get('tower_config_file'):
```

Comment:
The tower cli has a few default config file locations https://docs.ansible.com/ansible-tower/3.5.3/html/towerapi/tower_cli.html#file-locations , we probably need to be backwards compatible with that

Reply:
This should be resolved now. We need to see if we can somehow clone the existing tower-cli/tests/test_config.py file.